### PR TITLE
[VL] Avoid initializing C++ backend instance more than once

### PR DESCRIPTION
**Pull request by zhztheplayer**
The patch includes changes as follow:

* Consolidate implementation code of `addRecycler` and `addResourceManager`
* Rename `TaskMemoryResources` to `TaskResources` (the same to other classes that start with `TaskMemory`)
* Move memory cap / memory threshold initializations from `VeloxBackend` to `VeloxInitializer`
* Move `ExpressionEvaluatorJniWrapper` and `NativeExpressionEvaluator` from `gluten-core` to `backends-velox` and `backend-clickhouse` (the JNI interface has already become different on these two backends for sometime)
* Add `InitializerJniWrapper`, and make ExpressionEvaluatorJniWrapper away from the process-wise backend initialization procedure
* Remove some unused codes from `JniCommon.h`
* Add a JNI API to track on the Spark task context lifecycle

### CODE REVIEW

File: cpp/velox/compute/VeloxInitializer.cc [hunk 1]
```diff
@@ -40,29 +41,95 @@ DECLARE_int32(split_preload_per_driver);
...
+
+// memory cache
+const std::string kVeloxMemCacheSize = "spark.gluten.sql.columnar.backend.velox.memCacheSize";
+const std::string kVeloxMemCacheSizeDefault = "1073741824";
+
+// ssd cache
+const std::string kVeloxSsdCacheSize = "spark.gluten.sql.columnar.backend.velox.ssdCacheSize";
+const std::string kVeloxSsdCacheSizeDefault = "1073741824";
+const std::string kVeloxSsdCachePath = "spark.gluten.sql.columnar.backend.velox.ssdCachePath";
+const std::string kVeloxSsdCachePathDefault = "/tmp/";
```
Comment (FelixYBW) [inline cpp/velox/compute/VeloxInitializer.cc:57]: not related to the PR. 
We mentioned to change the ssd cache from /tmp/ to the shuffle dirs as spark_local_dirs. Do we have an issue to track this? If not, let's create one

**Comment by zhztheplayer** [comment]
@rui-mo @zzcclp

**Comment by rui-mo** [comment]
> Move ExpressionEvaluatorJniWrapper and NativeExpressionEvaluator from gluten-core to backends-velox and backend-clickhouse (the JNI interface has already become different on these two backends for sometime)

Do we intend to use unified JNI interface or not in the future? cc @zzcclp

**Review by rui-mo — APPROVED** [review]

**Review by zzcclp — APPROVED** [review]
LGTM

**Comment by zhztheplayer** [comment]
Thanks!

**Pull request merged**